docs-samples/V4/JS/add-suggested-actions/app.js

```diff
@@ -0,0 +1,48 @@
+/*
+ * Botbuilder v4 SDK - Add suggested actions
+ * 
+ * This bot shows buttons the user can press to provide input.
+ * 
+ * To run this bot:
+ * 1) install these npm packages:
+ * npm install --save restify
+ * npm install --save botbuilder@preview
+ * 
+ * 2) From VSCode, open the package.json file and
+ * Update the property "main" to the name of the sample bot you want to run. 
+ *    For example: "main": "add-suggested-actions/app.js" to run the this sample bot.
+ * 3) run the bot in debug mode. 
+ * 4) Load the emulator and point it to: http://localhost:3978/api/messages
+ * 5) Send the message "hi" to engage with the bot to see the suggested actions.
+ *
+ */ 
+
+const { BotFrameworkAdapter, MessageFactory } = require('botbuilder');
+const restify = require('restify');
+
+// Create server
+let server = restify.createServer();
+server.listen(process.env.port || process.env.PORT || 3978, function () {
+    console.log(`${server.name} listening to ${server.url}`);
+});
+
+// Create adapter
+const adapter = new BotFrameworkAdapter({ 
+    appId: process.env.MICROSOFT_APP_ID, 
+    appPassword: process.env.MICROSOFT_APP_PASSWORD 
+});
+
+
+// Listen for incoming requests 
+server.post('/api/messages', (req, res) => {
+    // Route received request to adapter for processing
+    adapter.processActivity(req, res, async (context) => {
+        if (context.activity.type === 'message') {
+
+            //  Initialize the message object.
+            const basicMessage = MessageFactory.suggestedActions(['red', 'green', 'blue'], 'Choose a color');
+
+            await context.sendActivity(basicMessage);
```

Comment:
Since it's action driven, I suggest you restructure this sample to behave like a prompt...or an event driven format where you present the buttons, then handle the click event when the user chooses one of the button. You could simply reiterate the user's choice to them (for simplicity) but at least it will show the real use of buttons in suggested actions.